[Log Starts]
ERROR : 2016-08-30 16:10:55,978 19281ms AccountHelper          UpdateUser         - Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=472540 for information on understanding and handling optimistic concurrency exceptions.System.Data.Entity.Infrastructure.DbUpdateConcurrencyException: Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=472540 for information on understanding and handling optimistic concurrency exceptions. ---> System.Data.Entity.Core.OptimisticConcurrencyException: Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=472540 for information on understanding and handling optimistic concurrency exceptions.
   at System.Data.Entity.Core.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Entity.Core.Mapping.Update.Internal.UpdateTranslator.Update()
   at System.Data.Entity.Core.EntityClient.Internal.EntityAdapter.<Update>b__2(UpdateTranslator ut)
   at System.Data.Entity.Core.EntityClient.Internal.EntityAdapter.Update[T](T noChangesResult, Func`2 updateFunction)
   at System.Data.Entity.Core.EntityClient.Internal.EntityAdapter.Update()
   at System.Data.Entity.Core.Objects.ObjectContext.<SaveChangesToStore>b__35()
   at System.Data.Entity.Core.Objects.ObjectContext.ExecuteInTransaction[T](Func`1 func, IDbExecutionStrategy executionStrategy, Boolean startLocalTransaction, Boolean releaseConnectionOnSuccess)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChangesToStore(SaveOptions options, IDbExecutionStrategy executionStrategy, Boolean startLocalTransaction)
   at System.Data.Entity.Core.Objects.ObjectContext.<>c__DisplayClass2a.<SaveChangesInternal>b__27()
   at System.Data.Entity.SqlServer.DefaultSqlExecutionStrategy.Execute[TResult](Func`1 operation)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChangesInternal(SaveOptions options, Boolean executeInExistingTransaction)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   --- End of inner exception stack trace ---
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at UI.Helpers.AccountHelper.UpdateUser(UserProfile user) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Helpers\AccountHelper.cs:line 67

[Log Ends]

[Log Starts]
ERROR : 2016-08-30 16:16:52,589 14484ms AccountHelper          UpdateUser         - Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=472540 for information on understanding and handling optimistic concurrency exceptions.System.Data.Entity.Infrastructure.DbUpdateConcurrencyException: Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=472540 for information on understanding and handling optimistic concurrency exceptions. ---> System.Data.Entity.Core.OptimisticConcurrencyException: Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=472540 for information on understanding and handling optimistic concurrency exceptions.
   at System.Data.Entity.Core.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Entity.Core.Mapping.Update.Internal.UpdateTranslator.Update()
   at System.Data.Entity.Core.EntityClient.Internal.EntityAdapter.<Update>b__2(UpdateTranslator ut)
   at System.Data.Entity.Core.EntityClient.Internal.EntityAdapter.Update[T](T noChangesResult, Func`2 updateFunction)
   at System.Data.Entity.Core.EntityClient.Internal.EntityAdapter.Update()
   at System.Data.Entity.Core.Objects.ObjectContext.<SaveChangesToStore>b__35()
   at System.Data.Entity.Core.Objects.ObjectContext.ExecuteInTransaction[T](Func`1 func, IDbExecutionStrategy executionStrategy, Boolean startLocalTransaction, Boolean releaseConnectionOnSuccess)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChangesToStore(SaveOptions options, IDbExecutionStrategy executionStrategy, Boolean startLocalTransaction)
   at System.Data.Entity.Core.Objects.ObjectContext.<>c__DisplayClass2a.<SaveChangesInternal>b__27()
   at System.Data.Entity.SqlServer.DefaultSqlExecutionStrategy.Execute[TResult](Func`1 operation)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChangesInternal(SaveOptions options, Boolean executeInExistingTransaction)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   --- End of inner exception stack trace ---
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at UI.Helpers.AccountHelper.UpdateUser(UserProfile user) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Helpers\AccountHelper.cs:line 70

[Log Starts]
ERROR : 2016-08-30 16:23:13,120 60645ms AccountHelper          UpdateUser         - Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=472540 for information on understanding and handling optimistic concurrency exceptions.System.Data.Entity.Infrastructure.DbUpdateConcurrencyException: Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=472540 for information on understanding and handling optimistic concurrency exceptions. ---> System.Data.Entity.Core.OptimisticConcurrencyException: Store update, insert, or delete statement affected an unexpected number of rows (0). Entities may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=472540 for information on understanding and handling optimistic concurrency exceptions.
   at System.Data.Entity.Core.Mapping.Update.Internal.UpdateTranslator.ValidateRowsAffected(Int64 rowsAffected, UpdateCommand source)
   at System.Data.Entity.Core.Mapping.Update.Internal.UpdateTranslator.Update()
   at System.Data.Entity.Core.EntityClient.Internal.EntityAdapter.<Update>b__2(UpdateTranslator ut)
   at System.Data.Entity.Core.EntityClient.Internal.EntityAdapter.Update[T](T noChangesResult, Func`2 updateFunction)
   at System.Data.Entity.Core.EntityClient.Internal.EntityAdapter.Update()
   at System.Data.Entity.Core.Objects.ObjectContext.<SaveChangesToStore>b__35()
   at System.Data.Entity.Core.Objects.ObjectContext.ExecuteInTransaction[T](Func`1 func, IDbExecutionStrategy executionStrategy, Boolean startLocalTransaction, Boolean releaseConnectionOnSuccess)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChangesToStore(SaveOptions options, IDbExecutionStrategy executionStrategy, Boolean startLocalTransaction)
   at System.Data.Entity.Core.Objects.ObjectContext.<>c__DisplayClass2a.<SaveChangesInternal>b__27()
   at System.Data.Entity.SqlServer.DefaultSqlExecutionStrategy.Execute[TResult](Func`1 operation)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChangesInternal(SaveOptions options, Boolean executeInExistingTransaction)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChanges(SaveOptions options)
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   --- End of inner exception stack trace ---
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at System.Data.Entity.Internal.LazyInternalContext.SaveChanges()
   at System.Data.Entity.DbContext.SaveChanges()
   at UI.Helpers.AccountHelper.UpdateUser(UserProfile user) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Helpers\AccountHelper.cs:line 70

[Log Starts]
[Log Ends]

[Log Starts]
ERROR : 2016-08-30 16:33:23,283 14403ms DataAccess             CreateDataSet      - The current TransactionScope is already complete.System.InvalidOperationException: The current TransactionScope is already complete.
   at System.Transactions.Transaction.get_Current()
   at System.Data.ProviderBase.DbConnectionPool.GetFromTransactedPool(Transaction& transaction)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at System.Data.Common.DbDataAdapter.FillInternal(DataSet dataset, DataTable[] datatables, Int32 startRecord, Int32 maxRecords, String srcTable, IDbCommand command, CommandBehavior behavior)
   at System.Data.Common.DbDataAdapter.Fill(DataSet dataSet, Int32 startRecord, Int32 maxRecords, String srcTable, IDbCommand command, CommandBehavior behavior)
   at System.Data.Common.DbDataAdapter.Fill(DataSet dataSet)
   at DAL.DataAccess.CreateDataSet(String connString, String sql, CommandType commandType, String datasetName, SqlParameter[] parameters) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\DAL\DataAccess.cs:line 125

ERROR : 2016-08-30 16:33:23,306 14426ms AccountHelper          UpdateUser         - The current TransactionScope is already complete.System.Exception: The current TransactionScope is already complete.
   at DAL.DataAccess.CreateDataSet(String connString, String sql, CommandType commandType, String datasetName, SqlParameter[] parameters) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\DAL\DataAccess.cs:line 136
   at UI.Helpers.AccountHelper.GetRoleListByUserId(Int32 userId) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Helpers\AccountHelper.cs:line 187
   at UI.Models.UserEditModel.get_SelectedRoles() in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Models\UserViewModel.cs:line 55
   at UI.Helpers.AccountHelper.UpdateUser(UserEditModel model) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Helpers\AccountHelper.cs:line 72

[Log Ends]

[Log Starts]
ERROR : 2016-08-30 16:34:16,838 10920ms AccountHelper          UpdateUser         - MSDTC on server 'KEVINLAU\SQLEXPRESS' is unavailable.System.Data.SqlClient.SqlException (0x80131904): MSDTC on server 'KEVINLAU\SQLEXPRESS' is unavailable.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.TdsParser.TdsExecuteTransactionManagerRequest(Byte[] buffer, TransactionManagerRequestType request, String transactionName, TransactionManagerIsolationLevel isoLevel, Int32 timeout, SqlInternalTransaction transaction, TdsParserStateObject stateObj, Boolean isDelegateControlRequest)
   at System.Data.SqlClient.TdsParser.GetDTCAddress(Int32 timeout, TdsParserStateObject stateObj)
   at System.Data.SqlClient.SqlInternalConnectionTds.GetDTCAddress()
   at System.Data.SqlClient.SqlInternalConnection.EnlistNonNull(Transaction tx)
   at System.Data.SqlClient.SqlInternalConnection.Enlist(Transaction tx)
   at System.Data.SqlClient.SqlInternalConnectionTds.Activate(Transaction transaction)
   at System.Data.ProviderBase.DbConnectionInternal.ActivateConnection(Transaction transaction)
   at System.Data.ProviderBase.DbConnectionPool.PrepareConnection(DbConnection owningObject, DbConnectionInternal obj, Transaction transaction)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at WebMatrix.Data.Database.EnsureConnectionOpen()
   at WebMatrix.Data.Database.<QueryInternal>d__0.MoveNext()
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source)
   at WebMatrix.Data.Database.QuerySingle(String commandText, Object[] args)
   at WebMatrix.WebData.DatabaseWrapper.QuerySingle(String commandText, Object[] parameters)
   at WebMatrix.WebData.SimpleRoleProvider.IsUserInRole(String username, String roleName)
   at WebMatrix.WebData.SimpleRoleProvider.AddUsersToRoles(String[] usernames, String[] roleNames)
   at System.Web.Security.Roles.AddUserToRole(String username, String roleName)
   at UI.Helpers.AccountHelper.UpdateUser(UserEditModel model) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Helpers\AccountHelper.cs:line 72
ClientConnectionId:a03522bb-f031-426c-8cc4-e7b3aff29114
Error Number:8501,State:2,Class:16

[Log Ends]

[Log Starts]
ERROR : 2016-08-30 16:43:10,825 20437ms AccountHelper          UpdateUser         - User "tests" is already in role "Admin".System.InvalidOperationException: User "tests" is already in role "Admin".
   at WebMatrix.WebData.SimpleRoleProvider.AddUsersToRoles(String[] usernames, String[] roleNames)
   at System.Web.Security.Roles.AddUserToRole(String username, String roleName)
   at UI.Helpers.AccountHelper.UpdateUser(UserEditModel model) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Helpers\AccountHelper.cs:line 69

[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
ERROR : 2016-08-30 17:53:16,562 12699ms AccountHelper          UpdateUser         - User "tests" is not in role "General User".System.InvalidOperationException: User "tests" is not in role "General User".
   at WebMatrix.WebData.SimpleRoleProvider.RemoveUsersFromRoles(String[] usernames, String[] roleNames)
   at System.Web.Security.Roles.RemoveUserFromRoles(String username, String[] roleNames)
   at UI.Helpers.AccountHelper.UpdateUser(UserEditModel model) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Helpers\AccountHelper.cs:line 68

[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
ERROR : 2016-08-30 19:34:44,708 14326ms AccountHelper          UpdateUser         - The array parameter 'roleNames' should not be empty.
Parameter name: roleNamesSystem.ArgumentException: The array parameter 'roleNames' should not be empty.
Parameter name: roleNames
   at System.Web.Util.SecUtility.CheckArrayParameter(String[]& param, Boolean checkForNull, Boolean checkIfEmpty, Boolean checkForCommas, Int32 maxSize, String paramName)
   at System.Web.Security.Roles.RemoveUserFromRoles(String username, String[] roleNames)
   at UI.Helpers.AccountHelper.UpdateUser(UserEditModel model) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Helpers\AccountHelper.cs:line 68

ERROR : 2016-08-30 19:35:45,284 74902ms AccountHelper          UpdateUser         - The array parameter 'roleNames' should not be empty.
Parameter name: roleNamesSystem.ArgumentException: The array parameter 'roleNames' should not be empty.
Parameter name: roleNames
   at System.Web.Util.SecUtility.CheckArrayParameter(String[]& param, Boolean checkForNull, Boolean checkIfEmpty, Boolean checkForCommas, Int32 maxSize, String paramName)
   at System.Web.Security.Roles.RemoveUserFromRoles(String username, String[] roleNames)
   at UI.Helpers.AccountHelper.UpdateUser(UserEditModel model) in E:\Kevin Lau\Documents\Visual Studio 2013\Projects (E)\WordCollocation2016_wc\UI\Helpers\AccountHelper.cs:line 68

[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

[Log Starts]
[Log Ends]

